src/main/java/core/basesyntax/operation/BalanceOperation.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.merge(
+                transaction.getFruit(),
+                transaction.getQuantity(),
+                Integer::sum
+        );
+    }
```

Comment:
The implementation of the `apply` method in `BalanceOperation` should not use `Integer::sum` as it's meant to set the balance to the specific value from the transaction, not add to the existing value. Consider setting the value directly instead of merging it.